scene/resources/shader.h

```diff
@@ -87,15 +87,41 @@ class Shader : public Resource {
 
 	virtual bool is_text_shader() const;
 
-	_FORCE_INLINE_ StringName remap_uniform(const StringName &p_uniform) const {
+	// Finds the shader parameter name for the given property name, which should start with "shader_param/".
+	_FORCE_INLINE_ StringName remap_parameter(const StringName &p_property) const {
 		if (params_cache_dirty) {
 			get_shader_uniform_list(nullptr);
 		}
 
-		const HashMap<StringName, StringName>::Iterator E = params_cache.find(p_uniform);
-		if (E) {
-			return E->value;
+		String n = p_property;
+
+		// Backwards compatibility with old shader parameter names.
+		// Note: These checks are needed, since otherwise shader_param/ would turn into shader_shader_param/
+		if (n.find("param/") == 0) {
+			n = n.replace_first("param/", "shader_param/");
+		}
+		if (n.find("shader_uniform/") == 0) {
+			n = n.replace_first("shader_uniform/", "shader_param/");
+		}
+
+		{
+			// Additional backwards compatibility for projects between #62972 and #64092 (about a month of v4.0 development).
+			// These projects did not have any prefix for shader uniforms due to a bug.
+			// This code should be removed during beta or rc of 4.0.
```

Comment:
CC @tinmanjuggernaut

How long should this compatibility last? I put in the comment beta or rc 4.0, but beta is already nearing close. It might be worth it to keep it for now, then remove _all_ compatibility in Godot 5 (shouldn't be a big deal, since I think `shader_param/` is now used in Godot 3, 4, and future versions)